activerecord/test/cases/transaction_callbacks_test.rb

```diff
@@ -83,6 +83,16 @@ def test_call_after_commit_after_transaction_commits
     assert_equal [:after_commit], @first.history
   end
 
+  def test_update_in_after_commit_on_create
+    new_record = TopicWithCallbacks.new(title: 'Topic')
+    new_record.after_commit_block(:create) do |r|
```

Comment:
Not sure if I am missing something, but isn't this error about `after_commit on: :create` been called on update. Thats exactly what your test is doing, (the wrong behaviour)